roles/translator/src/downstream_sv1/downstream.rs

```diff
@@ -104,26 +104,30 @@ impl Downstream {
         // role, or the message is sent upwards to the Bridge for translation into a SV2 message
         // and then sent to the SV2 Upstream role.
         let _socket_reader_task = task::spawn(async move {
+            let mut messages = BufReader::new(&*socket_reader).lines();
```

Comment:
I don't think it's new with this commit, but I believe this is vulnerable to DOS attacks since `lines()` will read indefinitely, storing all read bytes if no line terminator is found (potential out-of-memory).

Reply:
Isn't it bounded by the capacity from the BufReader::new() which has a default capacity of 1024?

pub fn new(inner: R) -> BufReader<R> {
        BufReader::with_capacity(DEFAULT_BUF_SIZE, inner)
    }

Reply:
The capacity will make it so the Lines aggregator will only read 8 kb at a time, but the aggregator itself has no cap as far as I can tell:
https://github.com/async-rs/async-std/blob/main/src/io/buf_read/lines.rs#L64
https://github.com/async-rs/async-std/blob/main/src/io/buf_read/mod.rs#L242-L266